File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string str) {
        
      map<char,int>mp;
        
        mp['I']=1;
        mp['V']=5;
        mp['X']=10;
        mp['L']=50;
        mp['C']=100;
        mp['D']=500;
        mp['M']=1000;
        
        int n=str.length();
        
        char c=str.back();
        int sum=mp[c];
        for(int i=n-2;i>=0;i--)
        {
            if(mp[str[i]]>=mp[c])
            {
                sum+=mp[str[i]];
                
            }
            else{
                sum-=mp[str[i]];
            }
            c=str[i];
        }
        return sum;
      
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        
        mp1 = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
        
        total=mp1[s[-1]]
        
        for i in range(len(s)-2,-1,-1):
            
            if(mp1[s[i]]>=mp1[s[i+1]]):
                total+=mp1[s[i]]
                
            else:
                total-=mp1[s[i]]
                
                
        return total       
                
                
            
    
        